### DISCUSSION TITLE
How to get authenticated while testing

### DISCUSSION THREAD

**Discussion by bhubavenski** [nextauthjs/next-auth#12130] [Help]
I am trying to test my app, while I am authenticated. For authentication I am using next-auth V4. What many people are doing is to manually attach specific cookies and etc. so they can get authenticated. I don't know what I have to do. I am using next auth and I don't know what it is looking for so I can provide it. So, how I can test my app, as authenticated person, without going to /login page every time before start testing the real thing

**Comment by khuezy** [comment]
Testing locally or for your unit/e2e test? You can use the `encode` method from `next-auth/jwt` then manually add that token to your browser cookie: `next-auth.session-token: <encoded jwt>`. Or just setup a test oauth provider, eg google and sign in.

**Comment by bhubavenski** [comment]
Ok, but, what params to set in “encode” function
